dbt/compat.py

```diff
@@ -43,10 +44,18 @@ def to_string(s):
             return str(s)
 
 
-def write_file(path, s):
+def _open(path, mode):
     if WHICH_PYTHON == 2:
-        with codecs.open(path, 'w', encoding='utf-8') as f:
-            return f.write(to_string(s))
+        return codecs.open(path, mode, encoding='utf-8')
     else:
-        with open(path, 'w') as f:
-            return f.write(to_string(s))
+        return open(path, 'w')
+
+
+def write_file(path, s):
+    with _open(path, 'w') as fp:
+        return fp.write(to_string(s))
+
+
+def write_json(path, data, **kwargs):
+    with _open(path, 'w') as fp:
+        json.dump(data, fp, **kwargs)
```

Comment:
i don't love this as part of the public API of dbt.compat, but i see the need for it. is there an advantage to doing it this way, vs calling `write_file` with `json.dumps(...)` from dbt.clients.system?

Reply:
Performance is different. `json.dump()` uses less memory because it streams, but can take longer to run because each 'chunk' of json becomes a `file.write()` call and it bypasses the "one shot" encoder in C. I don't know which is faster in any given situation, I'll just change it to use `json.dumps()` and hope we never run out of memory, we'll probably gain a lot from using the good encoder anyway.